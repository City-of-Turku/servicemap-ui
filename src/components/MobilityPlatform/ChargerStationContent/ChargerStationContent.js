import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { Typography } from '@material-ui/core';

const ChargerStationContent = ({ classes, intl, station }) => {
  const locale = useSelector(state => state.user.locale);

  const titleTypo = (messageId, props = {}) => (
    <div {...props}>
      <Typography variant="subtitle2">
        {intl.formatMessage({
          id: messageId,
        })}
        :
      </Typography>
    </div>
  );

  const singleValTypo = (messageId, value, props = {}) => (
    <div {...props}>
      <Typography>
        <strong>
          {intl.formatMessage({
            id: messageId,
          })}
          :
        </strong>
        {' '}
        {value}
      </Typography>
    </div>
  );

  const renderName = (nameFi, nameEn, nameSv) => {
    switch (locale) {
      case 'en':
        return nameEn;
      case 'sv':
        return nameSv;
      default:
        return nameFi;
    }
  };

  const renderAddress = () => {
    switch (locale) {
      case 'en':
        return singleValTypo('mobilityPlatform.content.address', station.address_en, { className: classes.margin });
      case 'sv':
        return singleValTypo('mobilityPlatform.content.address', station.address_sv, { className: classes.margin });
      default:
        return singleValTypo('mobilityPlatform.content.address', station.address_fi, { className: classes.margin });
    }
  };

  const renderAdministrator = (item) => {
    switch (locale) {
      case 'en':
        return singleValTypo('mobilityPlatform.chargerStations.content.admin', item.en, { className: classes.margin });
      case 'sv':
        return singleValTypo('mobilityPlatform.chargerStations.content.admin', item.sv, { className: classes.margin });
      default:
        return singleValTypo('mobilityPlatform.chargerStations.content.admin', item.fi, { className: classes.margin });
    }
  };

  const gasFillingInfo = (
    <>
      {titleTypo('mobilityPlatform.content.gfsTitle')}
      {singleValTypo('mobilityPlatform.content.address', station.address)}
      {singleValTypo('mobilityPlatform.content.gfsType', station.extra.lng_cng)}
      {singleValTypo('mobilityPlatform.content.operator', station.extra.operator)}
    </>
  );

  const chargerStationInfo = (
    <>
      {renderAddress()}
      {renderAdministrator(station.extra.administrator)}
      {titleTypo('mobilityPlatform.content.chargersTitle', { className: classes.margin })}
      {station.extra.chargers && station.extra.chargers.length > 0
        ? station.extra.chargers.map(charger => (
          <div key={charger.plug} className={classes.contentInner}>
            {singleValTypo('mobilityPlatform.content.cgsType', charger.plug)}
            {singleValTypo('mobilityPlatform.content.count', charger.number)}
            <Typography>
              <strong>
                {intl.formatMessage({
                  id: 'mobilityPlatform.content.power',
                })}
                :
              </strong>
              {' '}
              {charger.power}
              {' '}
              kW
            </Typography>
          </div>
        ))
        : null}
    </>
  );

  return (
    <div className={classes.container}>
      <div className={classes.headerContainer}>
        <Typography variant="subtitle1">
          {station.content_type.type_name === 'GFS'
            ? station.name
            : renderName(station.name, station.name_en, station.name_sv)}
        </Typography>
      </div>
      <div className={classes.textContainer}>
        {station.content_type.type_name === 'GFS' ? gasFillingInfo : chargerStationInfo}
      </div>
    </div>
  );
};

ChargerStationContent.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  intl: PropTypes.objectOf(PropTypes.any).isRequired,
  station: PropTypes.objectOf(PropTypes.any),
};

ChargerStationContent.defaultProps = {
  station: {},
};

export default ChargerStationContent;
