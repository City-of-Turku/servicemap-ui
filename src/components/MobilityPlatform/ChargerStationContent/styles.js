const styles = {
  container: {
    margin: '0.8rem',
  },
  headerContainer: {
    width: '85%',
    borderBottom: '1px solid #000',
    paddingBottom: '0.5rem',
  },
  headerText: {
    marginBottom: '0.4rem',
  },
  textContainer: {
    marginTop: '0.7rem',
  },
};

export default styles;
