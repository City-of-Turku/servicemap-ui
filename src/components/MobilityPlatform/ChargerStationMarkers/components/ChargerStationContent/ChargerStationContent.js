import { Typography } from '@mui/material';
import PropTypes from 'prop-types';
import React from 'react';
import styled from '@emotion/styled';
import { useIntl } from 'react-intl';
import {
  StyledContainer, StyledHeaderContainer, StyledTextContainer,
} from '../../../styled/styled';
import TextComponent from '../../../TextComponent';

const ChargerStationContent = ({ station }) => {
  const intl = useIntl();

  const titleTypo = messageId => (
    <StyledTextContainer>
      <Typography variant="subtitle2" component="h3">
        {intl.formatMessage({
          id: messageId,
        })}
        :
      </Typography>
    </StyledTextContainer>
  );

  const singleValTypo = (messageId, value) => (
    <StyledTextContainer>
      <Typography variant="body2">{intl.formatMessage({ id: messageId }, { value })}</Typography>
    </StyledTextContainer>
  );

  const stationName = {
    fi: station.name,
    en: station.name_en,
    sv: station.name_sv,
  };

  const stationAddress = {
    fi: station.address_fi,
    en: station.address_en,
    sv: station.address_sv,
  };

  const renderAdministrator = item => {
    const stationAdmin = {
      fi: item.fi,
      en: item.en,
      sv: item.sv,
    };

    return <TextComponent messageId="mobilityPlatform.chargerStations.content.admin" textObj={stationAdmin} />;
  };

  const renderPayment = paymentType => {
    const toLower = paymentType.toLowerCase();
    return (
      <StyledTextContainer>
        <Typography variant="body2">
          {intl.formatMessage({
            id:
              toLower === 'maksullinen'
                ? 'mobilityPlatform.chargerStations.content.charge'
                : 'mobilityPlatform.chargerStations.content.free',
          })}
        </Typography>
      </StyledTextContainer>
    );
  };

  // key property on .map() is long but it's only way to prevent all duplicate keys -warnings.
  const chargerStationInfo = (
    <>
      {station.address ? <TextComponent messageId="mobilityPlatform.content.address" textObj={stationAddress} /> : null}
      {station.extra.administrator.fi !== '' ? renderAdministrator(station.extra.administrator) : null}
      {renderPayment(station.extra.payment)}
      {titleTypo('mobilityPlatform.content.chargersTitle')}
      {station.extra.chargers && station.extra.chargers.length > 0
        ? station.extra.chargers.map(charger => (
          <StyledContentInner key={`${charger.plug}${charger.power}${charger.number}`}>
            {singleValTypo('mobilityPlatform.content.cgsType', charger.plug)}
            {singleValTypo('mobilityPlatform.content.count', charger.number)}
            <Typography variant="body2">
              {intl.formatMessage({ id: 'mobilityPlatform.content.power' }, { value: charger.power })}
            </Typography>
          </StyledContentInner>
        ))
        : null}
    </>
  );

  return (
    <StyledContainer>
      <StyledHeaderContainer>
        <TextComponent textObj={stationName} isTitle />
      </StyledHeaderContainer>
      <div>{chargerStationInfo}</div>
    </StyledContainer>
  );
};

const StyledContentInner = styled.div(({ theme }) => ({
  marginLeft: theme.spacing(1),
  marginBottom: theme.spacing(1),
}));

ChargerStationContent.propTypes = {
  station: PropTypes.shape({
    address: PropTypes.string,
    address_fi: PropTypes.string,
    address_en: PropTypes.string,
    address_sv: PropTypes.string,
    name: PropTypes.string,
    name_en: PropTypes.string,
    name_sv: PropTypes.string,
    extra: PropTypes.shape({
      payment: PropTypes.string,
      administrator: PropTypes.shape({
        fi: PropTypes.string,
      }),
      chargers: PropTypes.arrayOf(
        PropTypes.shape({
          plug: PropTypes.string,
          number: PropTypes.string,
          power: PropTypes.string,
        }),
      ),
    }),
  }),
};

ChargerStationContent.defaultProps = {
  station: {},
};

export default ChargerStationContent;
