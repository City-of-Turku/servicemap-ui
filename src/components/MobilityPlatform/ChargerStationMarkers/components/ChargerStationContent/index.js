import ChargerStationContent from './ChargerStationContent';

export default ChargerStationContent;
