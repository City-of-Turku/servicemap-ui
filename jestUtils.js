import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });