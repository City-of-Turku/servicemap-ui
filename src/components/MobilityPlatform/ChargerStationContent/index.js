import { withStyles } from '@material-ui/core';
import ChargerStationContent from './ChargerStationContent';
import styles from './styles';

export default withStyles(styles)(ChargerStationContent);
