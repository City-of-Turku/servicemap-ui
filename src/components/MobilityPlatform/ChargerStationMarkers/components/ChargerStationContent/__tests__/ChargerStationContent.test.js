/* eslint-disable max-len */
// Link.react.test.js
import React from 'react';
import ChargerStationContent from '../index';
import { initialState } from '../../../../../../redux/reducers/user';
import finnishTranslations from '../../../../../../i18n/fi';
import { getRenderWithProviders } from '../../../../../../../jestUtils';

const mockProps = {
  station: {
    name: 'Testinimi',
    name_en: 'Testname',
    name_sv: 'Testnamn',
    address: 'testiosoite',
    address_fi: 'osoite',
    address_en: 'address',
    address_sv: 'addres',
    extra: {
      payment: 'maksullinen',
      administrator: {
        fi: 'Testiylläpitäjä',
        en: 'Testadmin',
        sv: 'Testadmin',
      },
      chargers: [
        {
          plug: 'CCS',
          number: 4,
          power: 50,
        },
      ],
    },
  },
};

const renderWithProviders = getRenderWithProviders({
  user: initialState,
});

describe('<ChargerStationContent />', () => {
  it('should match snapshot', () => {
    const { container } = renderWithProviders(<ChargerStationContent {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<ChargerStationContent {...mockProps} />);

    const h6 = container.querySelectorAll('h6');
    const p = container.querySelectorAll('p');
    expect(h6[0].textContent).toEqual(mockProps.station.name);
    expect(h6[1].textContent).toEqual(`${finnishTranslations['mobilityPlatform.content.chargersTitle']}:`);
    expect(p[0].textContent).toEqual(`${finnishTranslations['mobilityPlatform.content.address']}: ${mockProps.station.address_fi}`);
    expect(p[1].textContent).toEqual(`${finnishTranslations['mobilityPlatform.chargerStations.content.admin']}: ${mockProps.station.extra.administrator.fi}`);
    expect(p[2].textContent).toEqual(`${finnishTranslations['mobilityPlatform.chargerStations.content.charge']}`);
    expect(p[3].textContent).toEqual(`${finnishTranslations['mobilityPlatform.content.cgsType']}: ${mockProps.station.extra.chargers[0].plug}`);
    expect(p[4].textContent).toEqual(`${finnishTranslations['mobilityPlatform.content.count']}: ${mockProps.station.extra.chargers[0].number}`);
    expect(p[5].textContent).toEqual('Teho: 50 kW');
  });
});
