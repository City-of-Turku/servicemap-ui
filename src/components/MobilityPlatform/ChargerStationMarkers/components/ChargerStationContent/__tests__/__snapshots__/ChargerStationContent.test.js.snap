// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChargerStationContent /> should match snapshot 1`] = `
<div>
  <div
    class="injectIntl(ChargerStationContent)-container-1"
  >
    <div
      class="injectIntl(ChargerStationContent)-headerContainer-2"
    >
      <div
        class="injectIntl(TextComponent)-margin-6"
      >
        <p
          class="MuiTypography-root MuiTypography-subtitle1"
        >
          Testinimi
        </p>
      </div>
    </div>
    <div
      class="injectIntl(ChargerStationContent)-textContainer-3"
    >
      <div
        class="injectIntl(TextComponent)-margin-6"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Osoite: osoite
        </p>
      </div>
      <div
        class="injectIntl(TextComponent)-margin-6"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Ylläpitäjä: Testiylläpitäjä
        </p>
      </div>
      <div
        class="injectIntl(ChargerStationContent)-margin-5"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Maksullinen
        </p>
      </div>
      <div
        class="injectIntl(ChargerStationContent)-margin-5"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2"
        >
          Latausasemat
          :
        </h6>
      </div>
      <div
        class="injectIntl(ChargerStationContent)-contentInner-4"
      >
        <div>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Sähkölatausaseman tyyppi: CCS
          </p>
        </div>
        <div>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Määrä: 4
          </p>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Teho: 50 kW
        </p>
      </div>
    </div>
  </div>
</div>
`;
